deploy.py

```diff
@@ -116,6 +117,13 @@ def image_requires_build(image_dir, commit_range=None):
 
     return image_touched
 
+def image_exists(image_spec):
+    try:
+        docker('pull', image_spec)
+        return True
+
+    except subprocess.CalledProcessError:
```

Comment:
@kcranston  i did merge this but can we provide a more specific message about why it failed here? like "oops i couldn't find the docker image to build hte hub?"